spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/BaseCassandraRDD.scala

```diff
@@ -0,0 +1,245 @@
+  *
+  * Configuration properties should be passed in the `SparkConf` configuration of `SparkContext`.
+  * `CassandraRDD` needs to open connection to Cassandra, therefore it requires appropriate connection property values
+  * to be present in `SparkConf`. For the list of required and available properties, see
+  * [[com.datastax.spark.connector.cql.CassandraConnector C a s s a n d r a C o n n e c t o r]].
+  *
+  * `CassandraRDD` divides the dataset into smaller partitions, processed locally on every cluster node.
+  * A data partition consists of one or more contiguous token ranges.
+  * To reduce the number of roundtrips to Cassandra, every partition is fetched in batches. The following
+  * properties control the number of partitions and the fetch size:
+  *
+  * - spark.cassandra.input.split.size:        approx number of Cassandra partitions in a Spark partition, default 100000
+  * - spark.cassandra.input.page.row.size:     number of CQL rows fetched per roundtrip, default 1000
+  *
+  * A `CassandraRDD` object gets serialized and sent to every Spark executor.
+  *
+  * By default, reads are performed at ConsistencyLevel.LOCAL_ONE in order to leverage data-locality and minimize network traffic.
+  * This read consistency level is controlled by the following property:
+  *
+  * - spark.cassandra.input.consistency.level: consistency level for RDD reads, string matching the ConsistencyLevel enum name.
+  *
+  * If a Cassandra node fails or gets overloaded during read, queries are retried to a different node.
+  */
+abstract class BaseCassandraRDD[R, P] private[connector](
+                                                          @transient sc: SparkContext,
+                                                          val connector: CassandraConnector,
+                                                          val keyspaceName: String,
+                                                          val tableName: String,
+                                                          val columnNames: ColumnSelector = AllColumns,
+                                                          val where: CqlWhereClause = CqlWhereClause.empty,
+                                                          val readConf: ReadConf = ReadConf(),
+                                                          val dependecyList: Seq[Dependency[_]] = Seq.empty)(
+                                                          implicit
+                                                          ct: ClassTag[R], rddt: ClassTag[P],
+                                                          @transient rtf: RowReaderFactory[R])
+  extends RDD[P](sc, dependecyList) with Logging {
+
+  /* Logging classes inheritance conflict fix. */
+  override def log = super[Logging].log
+
+  override def logName = super[Logging].logName
+
+  override def logInfo(msg: => String) = super[Logging].logInfo(msg)
```

Comment:
Yeah this is just from the old code, I"ll rebase to your mods if SPARKC-54 gets merged first